provide.go

```diff
@@ -411,25 +441,31 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) (err error) {
 	oldProviders := make(map[key][]*constructorNode)
 	for k := range keys {
 		// Cache old providers before running cycle detection.
-		oldProviders[k] = c.providers[k]
-		c.providers[k] = append(c.providers[k], n)
+		oldProviders[k] = s.providers[k]
+		s.providers[k] = append(s.providers[k], n)
 	}
 
-	c.isVerifiedAcyclic = false
-	if !c.deferAcyclicVerification {
-		if ok, cycle := graph.IsAcyclic(c.gh); !ok {
-			// When a cycle is detected, recover the old providers to reset
-			// the providers map back to what it was before this node was
-			// introduced.
-			for k, ops := range oldProviders {
-				c.providers[k] = ops
+	for _, s := range allScopes {
+		s.isVerifiedAcyclic = false
+		if !s.deferAcyclicVerification {
```

Comment:
We can reduce nesting here by returning early if s.deferAcyclicVerification is true. 

```
if s.deferAcyclicVerification { continue }
```